apps/web/app/[lang]/studies/[id]/components/StudyApplyContent.tsx

```diff
@@ -1,141 +1,211 @@
-import { P2, H4 } from 'ui/core';
+'use client';
+
+import { P2, H4, RadioGroup } from 'ui/core';
 import { useTranslations } from 'intl';
 import { ImageWithFallback } from '@components/utils/ImageWithFallback';
-import { AgeToText, GenderToText, getTypeformIds } from '../helpers';
+import { AgeToText, getTypeformIds, parseTypeformUrl } from '../helpers';
 import { StudyApplyProps } from '../Study.types';
 import { useSearchParams } from 'next/navigation';
 import { TypeForm } from 'ui/components';
+import { useEffect, useState } from 'react';
+import { StudyApplyContentSkeleton } from './StudyApplyContentSkeleton';
 
 export const StudyApplyContent: React.FC<StudyApplyProps> = ({ study }) => {
-  const t = useTranslations('studies.studyDetail');
+  const t = useTranslations('studies.studyDetail.preview');
   const tActions = useTranslations('actions');
   const tStudyTypes = useTranslations('studies.studyDetail.types');
+  const tGender = useTranslations('studies.studyDetail.gender');
   const searchParams = useSearchParams();
+  const countryIdParam = searchParams.get('country_id');
+  const campaignIdParam = searchParams.get('campaign_id');
+
+  const [loading, setLoading] = useState(true);
+  const [showCountrySelection, setShowCountrySelection] = useState(false);
+  const [typeformProps, setTypeformProps] = useState<{
```

Comment:
```suggestion
  const [typeformData, setTypeformData] = useState<{
```

We use "Props" to define component properties type